Bug: Report data failed to refresh on 12/6

**Describe the bug**

https://github.com/cal-itp/reports/actions/runs/19996369744

```
Run make data
poetry run python generate_reports_data.py

  0%|          | 0/4 [00:00<?, ?it/s]

2024:   0%|          | 0/12 [00:00<?, ?it/s]


08:   0%|          | 0/180 [00:00<?, ?it/s]


                                           

                                            
  0%|          | 0/4 [00:09<?, ?it/s]
Disposing of database engine...
Database engine disposed.
Traceback (most recent call last):

  File "/home/runner/work/reports/reports/reports/generate_reports_data.py", line 565, in <module>
    typer.run(generate_data)

  File "/home/runner/work/reports/reports/reports/generate_reports_data.py", line 556, in generate_data
    generate_data_by_file_path(

  File "/home/runner/work/reports/reports/reports/generate_reports_data.py", line 498, in generate_data_by_file_path
    dump_report_data(

  File "/home/runner/work/reports/reports/reports/generate_reports_data.py", line 405, in dump_report_data
    service_hours = generate_daily_service_hours(itp_id, date_start, date_end)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/home/runner/work/reports/reports/reports/generate_reports_data.py", line 120, in generate_daily_service_hours
    _daily_service_hours()

  File "/home/runner/.cache/pypoetry/virtualenvs/reports-BegPBLPe-py3.11/lib/python3.11/site-packages/siuba/siu/calls.py", line 214, in __rrshift__
    return self(x)
           ^^^^^^^

  File "/home/runner/.cache/pypoetry/virtualenvs/reports-BegPBLPe-py3.11/lib/python3.11/site-packages/siuba/siu/calls.py", line 189, in __call__
    return getattr(inst, self.func)(*rest, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/opt/hostedtoolcache/Python/3.11.14/x64/lib/python3.11/functools.py", line 909, in wrapper
    return dispatch(args[0].__class__)(*args, **kw)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/home/runner/.cache/pypoetry/virtualenvs/reports-BegPBLPe-py3.11/lib/python3.11/site-packages/siuba/dply/verbs.py", line 439, in filter
    res = arg(__data) if callable(arg) else arg
          ^^^^^^^^^^^

  File "/home/runner/.cache/pypoetry/virtualenvs/reports-BegPBLPe-py3.11/lib/python3.11/site-packages/siuba/siu/calls.py", line 193, in __call__
    return f_op(inst, *rest, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/home/runner/.cache/pypoetry/virtualenvs/reports-BegPBLPe-py3.11/lib/python3.11/site-packages/pandas/core/ops/common.py", line 72, in new_method
    return method(self, other)
           ^^^^^^^^^^^^^^^^^^^

  File "/home/runner/.cache/pypoetry/virtualenvs/reports-BegPBLPe-py3.11/lib/python3.11/site-packages/pandas/core/arraylike.py", line 62, in __ge__
    return self._cmp_method(other, operator.ge)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/home/runner/.cache/pypoetry/virtualenvs/reports-BegPBLPe-py3.11/lib/python3.11/site-packages/pandas/core/series.py", line 6243, in _cmp_method
    res_values = ops.comparison_op(lvalues, rvalues, op)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/home/runner/.cache/pypoetry/virtualenvs/reports-BegPBLPe-py3.11/lib/python3.11/site-packages/pandas/core/ops/array_ops.py", line 287, in comparison_op
    res_values = comp_method_OBJECT_ARRAY(op, lvalues, rvalues)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/home/runner/.cache/pypoetry/virtualenvs/reports-BegPBLPe-py3.11/lib/python3.11/site-packages/pandas/core/ops/array_ops.py", line 75, in comp_method_OBJECT_ARRAY
    result = libops.scalar_compare(x.ravel(), y, op)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "pandas/_libs/ops.pyx", line 107, in pandas._libs.ops.scalar_compare

TypeError: '>=' not supported between instances of 'datetime.date' and 'str'

make: *** [Makefile:15: data] Error 1
Error: Process completed with exit code 2.
```

**To Reproduce**
Steps to reproduce the behavior:


**Expected behavior**



**Additional context**

--- a/reports/generate_reports_data.py
+++ b/reports/generate_reports_data.py
@@ -116,12 +116,16 @@ def _daily_service_hours():
 
 
 def generate_daily_service_hours(itp_id: int, date_start, date_end):
-    return (
-        _daily_service_hours()
-        >> filtr(_.calitp_itp_id == itp_id)
-        >> filtr(_.service_date >= date_start)
-        >> filtr(_.service_date <= date_end)
-    )
+    df = _daily_service_hours()
+    df["service_date"] = pd.to_datetime(df["service_date"])
+    date_start = pd.Timestamp(date_start)
+    date_end = pd.Timestamp(date_end)
+
+    return df[
+        (df["calitp_itp_id"] == itp_id)
+        & (df["service_date"] >= date_start)
+        & (df["service_date"] <= date_end)
+    ]
 
 
 @cache

